### ISSUE TITLE
How to schedule learning rate in pytorch_lightning

### ISSUE THREAD

**Issue by saahiluppal** [PyTorchLightning/pytorch-lightning#3795]
How to schedule learning rate in pytorch lightning
all i know is, learning rate is scheduled in configure_optimizer() function inside LightningModule

**Comment by SkafteNicki** [comment]
A couple of examples are shown [here](https://pytorch-lightning.readthedocs.io/en/latest/optimizers.html#learning-rate-scheduling).
Basically, return two list: first is your optimizer(s) and the second is your schedulers.
```
def configure_optimizers(self):
    optimizer = Adam(self.parameters(), lr=1e-3)
    scheduler = ReduceLROnPlateau(optimizer, ...)
    return [optimizer], [scheduler]
```
lightning will call the scheduler internally.
Tip: you can use the [LearningRateMonitor](https://pytorch-lightning.readthedocs.io/en/latest/callbacks.html#learning-rate-monitor) callback to log your learning rate during training.

**Comment by saahiluppal** [comment]
I got it. 
Thanks.

**Comment by bokutotu** [comment]
@SkafteNicki both links (https://pytorch-lightning.readthedocs.io/en/latest/optimizers.html#learning-rate-scheduling, https://pytorch-lightning.readthedocs.io/en/latest/callbacks.html#learning-rate-monitor) seem dead.

**Comment by SkafteNicki** [comment]
@bokutotu sorry the docs structure has been slightly changed since my original response, so here are the updated links:

https://pytorch-lightning.readthedocs.io/en/latest/common/optimizers.html#learning-rate-scheduling

https://pytorch-lightning.readthedocs.io/en/latest/extensions/generated/pytorch_lightning.callbacks.LearningRateMonitor.html#pytorch_lightning.callbacks.LearningRateMonitor

**Comment by amansouri3476** [comment]
@SkafteNicki The links seem dead again.

**Comment by SkafteNicki** [comment]
@amansouri3476 here are the updated links:

Learning rate scheduler:
https://pytorch-lightning.readthedocs.io/en/latest/common/optimization.html#id5

Learning rate monitor callback:
https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.callbacks.LearningRateMonitor.html#pytorch_lightning.callbacks.LearningRateMonitor

In general i recommend to use the search field in the documentation if you are looking for something specific.

**Comment by layumi** [comment]
The latest Learning rate monitor callback is at 
https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.LearningRateMonitor.html
